src/breadth_first_search/__init__.py

```diff
@@ -0,0 +1,21 @@
+from collections import deque
+
+
+def breadth_first_search(graph, root_node, fn):
+    if not graph:
+        return False
+
+    search_queue = deque(graph.get(root_node, []))
+    checked = {}
+
+    while search_queue:
+        node = search_queue.popleft()
+        if not checked.get(node):
+            if fn(node):
+                return True
+
+            children = graph.get(node)
+            if children:
+                search_queue.extend(children)
+                checked[node] = True
```

Comment:
You add node to `checked` if `children` isn't empty. So it is excellent chance to execute `fn` several times with same argument

Reply:
:shipit: